pkg/kube/multicluster/cluster.go

```diff
@@ -49,7 +49,7 @@ func (r *Cluster) Run() {
 		time.AfterFunc(features.RemoteClusterTimeout, func() {
 			if !r.initialSync.Load() {
 				log.Errorf("remote cluster %s failed to sync after %v", r.ID, features.RemoteClusterTimeout)
-				timeouts.Increment()
+				timeouts.With(clusterID.Value(string(r.ID))).Increment()
```

Comment:
can we remove/clean the stats when a cluster is removed?

Reply:
`remote_cluster_sync_timeouts_total` is a Metric of type `Sum`, which should count all occurrences of errors, maybe we shouldn't be clear about it? 
When doing alerts based on this metric, you should also use `rate` calculation, it should not affect, what do you think?

Reply:
The worst-case scenario, too much cluster removed and never add back, is there some memory leak?

Reply:
After deep thinking, there're not too much cluster in a real world, and it's safe to rollout istiod if there's a real memory issue.